OsmAnd/src/net/osmand/plus/auto/MapMarkersScreen.kt

```diff
@@ -0,0 +1,112 @@
+package net.osmand.plus.auto
+
+import android.text.SpannableString
+import android.text.Spanned
+import androidx.car.app.CarContext
+import androidx.car.app.Screen
+import androidx.car.app.model.*
+import androidx.car.app.navigation.model.PlaceListNavigationTemplate
+import androidx.core.graphics.drawable.IconCompat
+import net.osmand.data.LatLon
+import net.osmand.plus.OsmandApplication
+import net.osmand.plus.R
+import net.osmand.plus.mapmarkers.MapMarker
+import net.osmand.plus.utils.AndroidUtils
+import net.osmand.search.core.ObjectType
+import net.osmand.search.core.SearchResult
+import net.osmand.util.MapUtils
+
+class MapMarkersScreen(
+    carContext: CarContext,
+    private val settingsAction: Action,
+    private val surfaceRenderer: SurfaceRenderer) : Screen(carContext) {
+
+    override fun onGetTemplate(): Template {
+        val listBuilder = ItemList.Builder()
+        val app = app
+        val markers = app.mapMarkersHelper.mapMarkers
+        val location = app.settings.lastKnownMapLocation
+        val iconsCache = app.uiUtilities
+        for (marker in markers) {
+            val title = marker.getName(app)
+            val markerColor = MapMarker.getColorId(marker.colorIndex)
+            val iconDrawable = iconsCache.getIcon(R.drawable.ic_action_flag, markerColor)
+            val icon = CarIcon.Builder(
+                IconCompat.createWithBitmap(AndroidUtils.drawableToBitmap(iconDrawable))).build()
```

Comment:
Many bitmaps may lead to [crashes ](https://github.com/osmandapp/OsmAnd/issues/16158)because these items are passed as parcelable to android auto. If possible, we could use iconId+color, or try to limit list size, similar to[ this place ](https://github.com/osmandapp/OsmAnd/blob/610b065b732af8a97a032ca6f7f918aedb0ec9aa/OsmAnd/src/net/osmand/plus/auto/SearchScreen.java#L80)